### DISCUSSION TITLE
Objects having a "AreYouATwinBasicInstance" interface

### DISCUSSION THREAD

**Discussion by Kr00l** [twinbasic/twinbasic#1277] [General]
The user 'The trick' in vbForums pointed out that **every** VB6 object support below interface:

```
[ InterfaceId ("0B6C9465-D082-11CF-8B4F-00A0C90F2704") ]
Interface AreYouABasicInstance Extends stdole.IUnknown
End Interface
```

So, in VB6 we could check for `If TypeOf MyVB6Class Is AreYouABasicInstance Then` if it is a native VB6 object (Form, Class etc.) or not.

I tested it in tB and it fails of course.

I am **not** asking to emulate the same in tB, because that could lead to wrong conclusions and would destroy the unique way to discover if it is a real VB6 object or not.

Would it make sense for tB to make it's own `AreYouATwinBasicInstance` interface supporting in each inner object ? (Form, Class etc.)

This way the outside COM world would have a nice and quick way to test the object.
But for what purpose ? .. in the end that matters. However, just wanted to point this fact out for documentation.

**Comment by wqweto** [comment]
FYI, `AreYouABasicInstance` interface allows to check if VB6 undocumented "object info" metadata is available like this
```
    Dim pVtable         As Long
    GetMem4 ByVal ObjPtr(Me), pVtable
    Dim ptObjectInfo    As Long
    GetMem4 ByVal pVtable - 4&, ptObjectInfo
```
. . . as discussed [here](https://www.vbforums.com/showthread.php?898040-RESOLVED-Absolute-(overall)-count-of-VTable-Entries).

It seems to have a lot of useful info which is implemented both in IDE and when compiled.

**Comment by WaynePhillipsEA** [comment]
Just FYI, in the depths of WinNativeForms you'll find this interface that every tB class / Form etc implements:

```
[ InterfaceId ("D90F2800-D7B1-4F62-82E4-8A1B52CA53E4") ]
Private Interface ITwinBasicInternalClassSupport Extends stdole.IUnknown
    Sub OverrideDispatchHandler(ByVal DelegatedDispatchHandler As Object)
End Interface
```

**Reply by mburns08109** [reply]
Wouldn't that be a good place to put AddressOf() also?
